### PR TITLE
`CellExplorerSortingInterface` to get sampling frequency from new CellExplorer format

### PR DESCRIPTION
**Pull request by h-mayorquin**
As mentioned before, in the new CellExplorer format the sampling frequency is the in the same file where the spikes are. I made the changes in spikeinterface to take this into account https://github.com/SpikeInterface/spikeinterface/pull/1628 but we won't have a release in a while.

Meanwhile, we will need this temporary change to the interface so we can use it in the current conversions.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
Confusing that the tests `TestOpenEphysBinaryRecordingInterfaceVersion` is failing here.

File: src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py [hunk 1]
```diff
@@ -25,7 +25,31 @@ def __init__(self, file_path: FilePathType, verbose: bool = True):
 
         hdf5storage = get_package(package_name="hdf5storage")
 
-        super().__init__(spikes_matfile_path=file_path, verbose=verbose)
+        file_path = Path(file_path)
+
+        # Temporary hack to get sampling frequency from the spikes cellinfo file until next SI release.
+        is_new_cell_explorer_format = "spikes.cellinfo.mat" in file_path.name
```
Comment (CodyCBakerPhD) [inline src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py:31]: Is there any version information contained in the files that could help identify 'new' vs. 'old' more precisely?

File: src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py [hunk 2]
```diff
@@ -25,7 +25,31 @@ def __init__(self, file_path: FilePathType, verbose: bool = True):
...
+
+            except NotImplementedError:
+                matlab_file = h5py.File(name=file_path, mode="r")
+                spikes_mat = matlab_file["spikes"]
+                assert isinstance(spikes_mat, h5py.Group), f"field `spikes` must be a Group, not {type(spikes_mat)}!"
+
+            sampling_frequency = spikes_mat.get("sr", None)
+            sampling_frequency = (
+                sampling_frequency[()] if isinstance(sampling_frequency, h5py.Dataset) else sampling_frequency
+            )
```
Comment (CodyCBakerPhD) [inline src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py:50]: Just wondering - is this guaranteed to always cast properly as a `np.float64`?

File: src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py (hunk 1 above)
Comment (h-mayorquin) [reply src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py:31]: Sadly it does not seem to be:

https://cellexplorer.org/datastructure/data-structure-and-format/#spikes

But, I think this condition is redundant. I should just try to extract the sampling rate, the old format does not have it, the new one should.

File: src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py [hunk 3]
```diff
@@ -25,7 +25,31 @@ def __init__(self, file_path: FilePathType, verbose: bool = True):
...
+
+            try:
+                matlab_file = scipy.io.loadmat(file_name=str(file_path), simplify_cells=True)
+                spikes_mat = matlab_file["spikes"]
+                assert isinstance(spikes_mat, dict), f"field `spikes` must be a dict, not {type(spikes_mat)}!"
+
+            except NotImplementedError:
+                matlab_file = h5py.File(name=file_path, mode="r")
+                spikes_mat = matlab_file["spikes"]
+                assert isinstance(spikes_mat, h5py.Group), f"field `spikes` must be a Group, not {type(spikes_mat)}!"
```
[the remark this answers is not in the document]
Comment (h-mayorquin) [reply src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py:45]: You need this structure to be a group or a dict so you can fetch the sampling rate as a dict below.

Sure with more informative error, this what h5py throws:
```
file["spikes"]
KeyError: "Unable to open object (object 'spikes' doesn't exist)"
```

You want "spikes" not in the matlab file? Feel free to add a suggestion

File: src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py (hunk 2 above)
Comment (h-mayorquin) [reply src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py:50]: No, this extracts the data the h5py dataset no matter the nesting.

File: src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py [hunk 4]
```diff
@@ -25,7 +25,31 @@ def __init__(self, file_path: FilePathType, verbose: bool = True):
...
-        super().__init__(spikes_matfile_path=file_path, verbose=verbose)
+        file_path = Path(file_path)
+
+        # Temporary hack to get sampling frequency from the spikes cellinfo file until next SI release.
+        import h5py
+
+        try:
+            matlab_file = scipy.io.loadmat(file_name=str(file_path), simplify_cells=True)
+            if "spikes" not in matlab_file.keys():
+                raise ValueError(f"Matlab file  {file_path} does not contain `spikes` field")
```
Comment (CodyCBakerPhD) [inline src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py:36]: Yeah this is definitely more informative; minor change to type though, this is still a KeyError right?

```suggestion
                raise KeyError(f"CellExplorer file '{file_path}' does not contain 'spikes' field.")
```

File: src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py [hunk 5]
```diff
@@ -25,7 +25,31 @@ def __init__(self, file_path: FilePathType, verbose: bool = True):
...
+            spikes_mat = matlab_file["spikes"]
+            assert isinstance(spikes_mat, dict), f"field `spikes` must be a dict, not {type(spikes_mat)}!"
+
+        except NotImplementedError:
+            matlab_file = h5py.File(name=file_path, mode="r")
+            if "spikes" not in matlab_file.keys():
+                raise ValueError(f"Matlab file  {file_path} does not contain `spikes` field")
```
Comment (CodyCBakerPhD) [inline src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py:43]:
```suggestion
                raise KeyError(f"CellExplorer file '{file_path}' does not contain 'spikes' field.")
```

File: src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py (hunk 2 above)
Comment (CodyCBakerPhD) [reply src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py]: Doesn't SI need sampling frequency attribute to be of specific data type? I'm just saying a simple `float` cast never hurt anyone 😅

Comment (h-mayorquin) [reply src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py]: A simple float cast will not work if the sampling_frequency is of type Dataset as far as I remember.

Comment (CodyCBakerPhD) [reply src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py]: Gotcha, OK then

File: src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py (hunk 4 above)
Comment (h-mayorquin) [reply src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py]: Correct about KeyError.

**Pull request merged**